spylon_kernel/scala_magic.py

```diff
@@ -115,72 +166,61 @@ def eval(self, code, raw):
-
-        This will instantiate a scala interpreter prior to running the code.
-
-        The -e or --eval_output flag signals that the retval value expression will be used as code for the cell to be
-        evaluated by the host language.
-
-        Examples:
-            %%scala
-            val x = 42
+    def post_process(self, retval):
+        """Processes the output of one or stacked magics.
 
-            %%scala
-            import collections.mutable._
-            val y = mutable.Map.empty[Int, String]
+        Part of the MetaKernel API.
 
-            %%scala -e
-            retval = "'(this is code in the kernel language)"
+        Parameters
+        ----------
+        retval : any or None
+            Value from another magic stacked with this one in a cell
 
-            %%python -e
-            "'(this is code in the kernel language)"
+        Returns
+        -------
+        any
+            The received value if it's not None, otherwise the stored
+            `retval` of the last Scala code execution
         """
-        if self.code.strip():
-            if eval_output:
-                self.eval(self.code, False)
-                # self.code = str(self.env["retval"]) if ("retval" in self.env and
-                #                                         self.env["retval"] != None) else ""
-                self.retval = None
-                #self.env["retval"] = None
-                self.evaluate = True
-            else:
-                self.retval = self.eval(self.code, False)
-                #self.env["retval"] = None
-                self.evaluate = False
-
-    def post_process(self, retval):
         if retval is not None:
             return retval
         else:
             return self.retval
 
     def get_completions(self, info):
+        """Gets completions from the kernel based on the provided info.
+
+        Part of the MetaKernel API.
+
+        Parameters
+        ----------
+        info : dict
+            Information returned by `metakernel.parser.Parser.parse_code`
+            including `code`, `help_pos`, `start`, etc.
+
+        Returns
+        -------
+        list of str
+            Possible completions for the code
+        """
         intp = self._get_scala_interpreter()
-        c = intp.complete(info['code'], info['help_pos'])
+        completions = intp.complete(info['code'], info['help_pos'])
 
         # Find common bits in the middle
         def trim(prefix, completions):
-            """Due to the nature of scala's completer we get full method names.
-            We need to trim out the common pieces.  Try longest prefix first etc
+            """Due to the nature of Scala's completer we get full method names.
+            We need to trim out the common pieces. Try longest prefix first, etc.
             """
             potential_prefix = os.path.commonprefix(completions)
             for i in reversed(range(len(potential_prefix)+1)):
```

Comment:
Passing on this one for now. I'll need to create some examples to really document this well.